src/App.js

```diff
@@ -1,79 +1,138 @@
 import React from 'react';
 
-import { getAll, getById } from './api/phone'
+import {getAll, getById} from './api/phone'
 import Basket from './Basket'
 import Filter from './Filter'
 import Catalog from './Catalog'
-
+import Viewer from './Viewer'
 import './App.css';
 
 
 class App extends React.Component {
-  constructor(props) {
-    super(props);
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            phones: getAll(),
+            selectedPhone: null,
+            basketItems: [],
+        };
+    }
 
-    this.state = {
-      phones: getAll(),
-      selectedPhone: null,
-      basketItems: [],
+    addToBasket = (phoneId) => {
+        const items = [...this.state.basketItems];
+        const itemIndex = items.findIndex((el) => el.name === phoneId);
+        if (itemIndex > -1) {
+          items[itemIndex] = {
+              ...items[itemIndex],
+              count: items[itemIndex].count + 1
+          };
+          this.setState({
+              basketItems: items,
+          })
+        } else {
+           this.setState({
+               basketItems: [...items, {name: phoneId, count: 1}]
+           })
+        }
     };
-  }
+    removeFromBasket = (index) => {
+        const newBasket = [...this.state.basketItems];
 
-  render() {
-    return (
-      <div className="App">
-        <div className="container-fluid">
-          <div className="row">
-            <div className="col-md-2">
-              <Filter />
-              <Basket />
-            </div>
 
-            <div className="col-md-10">
-              { this.state.selectedPhone ? (
-                <Viewer
-                  phone={this.state.selectedPhone}
-                  onBack={() => {
-                    this.setState({
-                      selectedPhone: null,
-                    });
-                  }}
-                />
-              ) : (
-                <Catalog
-                  phones={this.state.phones}
-                  onPhoneSelected={(phoneId) => {
-                    this.setState({
-                      selectedPhone: getById(phoneId),
-                    });
-                  }}
-                />
-              ) }
-            </div>
-          </div>
-        </div>
-      </div>
-    );
-  }
-}
+        this.setState((prevState) => {
+            if (prevState.basketItems[index].count >= 2) {
+                newBasket[index].count = newBasket[index].count - 1;
+            } else {
+                newBasket.splice(index);
```

Comment:
```suggestion
                newBasket.splice(index, 1);
```

**deleteCount** Optional
> An integer indicating the number of elements in the array to remove from start.
If deleteCount is omitted, or if its value is equal to or larger than array.length - start (that is, if it is equal to or greater than the number of elements left in the array, starting at start), then all the elements from start to the end of the array will be deleted.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice
Now if you have 10 items in the basket and decide to delete the first one, all items will be deleted.
![image](https://user-images.githubusercontent.com/29413892/59908570-61a67100-9416-11e9-92c4-fea785fb27c4.png)